### DISCUSSION TITLE
[RFC] Allow to pass custom props to #__next HTML element

### DISCUSSION THREAD

**Discussion by yordis** [vercel/next.js#19262] [Ideas]
# Feature request

## Is your feature request related to a problem? Please describe.

As a programmer, I would like to pass a class to `#__next` HTML element in order to overwrite the styles of it.

## Describe the solution you'd like

```tsx
function MyApp() {
    return (
      <Html>
        <Head></Head>
        <body className="bg-gray-100 text-base">
          {/* Notice that Main is taking className */}
          <Main className="flex flex-col" />
          <NextScript />
        </body>
      </Html>
    );
  }
```

## Describe alternatives you've considered

Use global styles instead, in environments using Tailwind (as an example) this feels odd, and against the flexibility of such a package, worth saying that is just an example.

## Additional context

- https://github.com/vercel/next.js/pull/17348

**Comment by ghost** [comment]
I add classes to the #__next container on almost every Next.js project. I like this idea, but in addition it would be great to pass class names from the next.config.js too. That way we're not required to add a custom document.js.

**Reply by yordis** [reply]
As long as that is not the only way to do it, I am cool with it.

Personally prefer to be explicit about this stuff, moving things into `next.config.js` just because we can is a rabbit hole.

We never know when we go from "simple abstractions where you can compose things" to "config-driven-development" type of situation.

I don't know the right answer here.

**Comment by yordis** [comment]
Today while trying to improve Accessibility I came across an issue where I need to pass extra props.

![Screen Shot 2021-11-28 at 2 46 49 AM](https://user-images.githubusercontent.com/4237280/143734206-be8f92dc-1863-45d7-ad54-febbcd793cf5.png)
